File: Plugins/Aspose_Imaging_Java_for_Ruby/lib/asposeimagingjava/images/cropimages.rb
module Asposeimagingjava
  module CropImages
    def initialize()
        # Cropping by Shifts
        crop_by_shifts()

        # Cropping by Rectangle
        crop_by_rectangle()
    end

    def crop_by_shifts()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image (of type bmp) in an instance of Image class
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Before cropping, the image should be cached for better performance
        if !image.isCached()
            image.cacheData()
        end

        # Define shift values for all four sides
        left_shift = 10
        right_shift = 10
        top_shift = 10
        bottom_shift = 10

        # Based on the shift values, apply the cropping on image
        # Crop method will shift the image bounds toward the center of image
        image.crop(left_shift, right_shift, top_shift, bottom_shift)

        # Save the image to disk
        image.save(data_dir + "CropByShifts.jpg")

        # Display Status.
        puts "Cropped image by shifts successfully!"
    end

    def crop_by_rectangle()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Load an existing image (of type bmp) in an instance of Image class
        image = Rjb::import('com.aspose.imaging.Image').load(data_dir + "test.jpg")

        # Before cropping, the image should be cached for better performance
        if !image.isCached()
            image.cacheData()
        end

        # Create an instance of Rectangle class with desired size
        rectangle = Rjb::import('com.aspose.imaging.Rectangle').new(10, 10, 100, 100)

        # Perform the crop operation on object of Rectangle class
        image.crop(rectangle)

        # Save the image to disk
        image.save(data_dir + "crop_by_rectangle.jpg")

        # Display Status.
        puts "Cropped image by rectangle successfully!"
    end
  end
end

File: Plugins/Aspose_Imaging_Java_for_PHP/src/aspose/imaging/ManagingRasterFormats/CropImages.php
<?php
namespace Aspose\Imaging\ManagingRasterFormats;

use com\aspose\imaging\Image as Image;
use com\aspose\imaging\Rectangle as Rectangle;
class CropImages{

    public static function run($dataDir=null){

        # Cropping by Shifts
        CropImages::crop_by_shifts($dataDir);

        # Cropping by Rectangle
        CropImages::crop_by_rectangle($dataDir);
    }

    public static function crop_by_shifts($dataDir=null){

    # Load an existing image (of type bmp) in an instance of Image class
    $image=new Image();
    $image = $image->load($dataDir."test.jpg");

    # Before cropping, the image should be cached for better performance
    if (!$image->isCached()) {
    $image->cacheData();
    }

    # Define shift values for all four sides
    $left_shift = 10;
    $right_shift = 10;
    $top_shift = 10;
    $bottom_shift = 10;

    # Based on the shift values, apply the cropping on image
    # Crop method will shift the image bounds toward the center of image
    $image->crop($left_shift, $right_shift, $top_shift, $bottom_shift);

    # Save the image to disk
    $image->save($dataDir . "CropByShifts.jpg");

    # Display Status.
    print "Cropped image by shifts successfully!".PHP_EOL;
    }

    public static function crop_by_rectangle($dataDir=null){

    # Load an existing image (of type bmp) in an instance of Image class
    $image=new Image();
    $image = $image->load($dataDir . "test.jpg");

    # Before cropping, the image should be cached for better performance
    if (!$image->isCached()) {
    $image->cacheData();
    }

    # Create an instance of Rectangle class with desired size
    $rectangle = new Rectangle(10, 10, 100, 100);

    # Perform the crop operation on object of Rectangle class
    $image->crop($rectangle);

    # Save the image to disk
    $image->save($dataDir . "crop_by_rectangle.jpg");

    # Display Status.
    print "Cropped image by rectangle successfully!".PHP_EOL;

    }

}
?>